xrshim.js

```diff
@@ -1,31 +1,176 @@
 {
 	const REALAPI_URL = 'https://raw.githack.com/MozillaReality/webxr-ios-js/develop/dist/webxr.js';
 
-	window.navigator.xr = {
-		isSessionSupported: function(mode) {
-			// Note: We support only immersive-ar mode for now.
-			//       See https://github.com/MozillaReality/webxr-ios-js/pull/34#discussion_r334910337
-			return Promise.resolve(mode === 'immersive-ar');
-		},
-		requestSession: function(mode, opts) {
-			console.log("going to load from ", REALAPI_URL)
-			return new Promise((resolve, reject) => {
-				delete window.navigator.xr;
-				const script = document.createElement('script');
-				script.setAttribute('src', REALAPI_URL);
-				script.setAttribute('type', 'text/javascript');
-
-				let loaded = false;
-				const loadFunction = () => {
-					if (loaded) return;
-					loaded = true;
-					console.log("now the script is really loaded", navigator.xr);
-					navigator.xr.requestSession(mode, opts).then(resolve).catch(reject);
-				};
-				script.onload = loadFunction;
-				script.onreadystatechange = loadFunction;
-				document.getElementsByTagName("head")[0].appendChild(script);
-			});
+	const API = [
+		'XR',
+		'XRSession',
+		'XRRenderState',
+		'XRFrame',
+		'XRSpace',
+		'XRReferenceSpace',
+		'XRBoundedReferenceSpace',
+		'XRView',
+		'XRViewport',
+		'XRRigidTransform',
+		'XRPose',
+		'XRViewerPose',
+		'XRInputSource',
+		'XRInputSourceArray',
+		'XRWebGLLayer',
+		'XRSessionEvent',
+		'XRInputSourceEvent',
+		'XRInputSourcesChangeEvent',
+		'XRReferenceSpaceEvent'
+	];
+
+	// Note: Currently (10/28/2019) EventTarget can't be extended on Safari
+	//	 unlike FireFox and Chrome. So declaring our own here for now.
+	//       But we don't expose so "navigator.xr instanceof EventTarget" will end up
+	//       being false out of this scope. Any problems with this limitation?
+	class EventTarget {
```

Comment:
Are Classes declared inside a scope like this local?  Because we can't override global Classes later, right?

Reply:
Yes, local in this scope (if I'm right). Two reasons why I don't override global EventTarget
1. I'm not really sure if we should do. Some applications/libraries might rely on native EventTarget. Non-overriding may be less problematic
2. The official WebXR polyfill we use seems to use their local EventTarget and doesn't override global EventTarget neither. I wanted to follow that manner. Maybe less problematic.